roles/translator/src/lib/proxy_config.rs

```diff
@@ -15,6 +15,68 @@ pub struct ProxyConfig {
     pub upstream_difficulty_config: UpstreamDifficultyConfig,
 }
 
+pub struct UpstreamConfig {
+    address: String,
+    port: u16,
+    authority_pubkey: Secp256k1PublicKey,
+    difficulty_config: UpstreamDifficultyConfig,
+}
+
+impl UpstreamConfig {
+    pub fn new(
+        address: String,
+        port: u16,
+        authority_pubkey: Secp256k1PublicKey,
+        difficulty_config: UpstreamDifficultyConfig,
+    ) -> Self {
+        Self {
+            address,
+            port,
+            authority_pubkey,
+            difficulty_config,
+        }
+    }
+}
+
+pub struct DownstreamConfig {
+    address: String,
+    port: u16,
+    difficulty_config: DownstreamDifficultyConfig,
+}
+
+impl DownstreamConfig {
+    pub fn new(address: String, port: u16, difficulty_config: DownstreamDifficultyConfig) -> Self {
+        Self {
+            address,
+            port,
+            difficulty_config,
+        }
+    }
+}
+
+impl ProxyConfig {
+    pub fn new(
```

Comment:
Same warning here, it seems the new() is not used anywhere.
So this comment is related to the previous one of course

Reply:
the `new` functiones are meant to be used in #1066 

should we suppress the warnings with a `#[allow(dead_code)]` or is there a more elegant solution?

Reply:
oh I see @jbesraa already addressed the comments around warnings

https://github.com/stratum-mining/stratum/pull/1092#issuecomment-2301771700

overall I'm not too opinionated on this... we could try to find a more elegant solution for a future PR, or just suppress warnings

Reply:
aside from this, I think this PR is pretty much good to go

Reply:
> the `new` functiones are meant to be used in #1066
> 
> should we suppress the warnings with a `#[allow(dead_code)]` or is there a more elegant solution?
> 
> tbh `#[allow(dead_code)]` is fine for me, given that `roles` are just PoCs
> 
> we're getting similar warnings on #1095 and #1097, so if we go with this solution, we should bring it over to those PRs too

For me it's ok to leave it as it is. I just wanted to get more info on this, thank you!

Reply:
fixed the warnings. its just a matter of making the unused structs public which I intended doing in #1066